### PR TITLE
*: add more debug log in slow query

### PR DESCRIPTION
**Pull request by lamxTyler**
### What problem does this PR solve? 

Add more debug bug log for tidb.

### What is changed and how it works?

It adds sql plan and session variable of join in slow log and support dump history 
stats.

### Check List 

Tests 

 - Unit test

Code changes

 - None

Related changes

- None

### CODE REVIEW

File: planner/core/stringer.go [hunk 1]
```diff
@@ -226,3 +226,144 @@ func toString(in Plan, strs []string, idxs []int) ([]string, []int) {
...
+	case *PhysicalSelection:
+		str = fmt.Sprintf("Sel(%s) %.2f", x.Conditions, x.StatsCount())
+	case *PhysicalProjection:
+		str = "Projection"
+	case *PhysicalTopN:
+		str = fmt.Sprintf("TopN(%v,%d,%d) %.2f", x.ByItems, x.Offset, x.Count, x.StatsCount())
+	case *PhysicalTableDual:
+		str = "Dual"
+	case *PhysicalHashAgg:
+		str = "HashAgg" + fmt.Sprintf("%.2f", x.StatsCount())
```
Comment (XuHuaiyu) [inline planner/core/stringer.go]: leave a blank before all the `%.2f` ?

File: planner/core/stringer.go [hunk 2]
```diff
@@ -226,3 +226,144 @@ func toString(in Plan, strs []string, idxs []int) ([]string, []int) {
...
+			strs, idxs = toString(c, strs, idxs)
+		}
+	}
+
+	var str string
+	switch x := in.(type) {
+	case *CheckTable:
+		str = "CheckTable"
+	case *PhysicalIndexScan:
+		str = fmt.Sprintf("Index(%s.%s) %.2f", x.Table.Name.L, x.Index.Name.L, x.StatsCount())
```
Comment (XuHuaiyu) [inline planner/core/stringer.go]: Range is not needed?

File: planner/core/stringer.go [hunk 3]
```diff
@@ -226,3 +226,144 @@ func toString(in Plan, strs []string, idxs []int) ([]string, []int) {
...
+		last := len(idxs) - 1
+		idx := idxs[last]
+		children := strs[idx:]
+		strs = strs[:idx]
+		idxs = idxs[:last]
+		str = "Apply{" + strings.Join(children, "->") + "}" + fmt.Sprintf("%.2f", x.StatsCount())
+	case *PhysicalMaxOneRow:
+		str = "MaxOneRow"
+	case *PhysicalLimit:
+		str = "Limit" + fmt.Sprintf("%v\n", x.StatsCount())
```
Comment (XuHuaiyu) [inline planner/core/stringer.go]: `\n` is not needed?

File: planner/core/stringer.go (hunk 2 above)
Comment (lamxTyler) [reply planner/core/stringer.go]: No, it may be too long and can be inferred from the sql.

**Comment by XuHuaiyu** [comment]
LGTM

**Pull request closed without merging**